radar-upload-backend/src/main/java/org/radarbase/upload/doa/RecordRepositoryImpl.kt

```diff
@@ -164,29 +179,37 @@ class RecordRepositoryImpl(@Context private var em: javax.inject.Provider<Entity
                 .resultList.firstOrNull()?.binaryStream?.readAllBytes()
     }
 
-    override fun create(record: Record): Record = em.get().transact {
-        val tmpContent = record.contents?.also { record.contents = null }
-
+    override fun create(record: Record, metadata: RecordMetadata?, contents: Set<ContentsDTO>?): Record = em.get().transact {
         persist(record)
 
-        record.apply {
-            contents = tmpContent?.mapTo(HashSet()) {
-                it.record = record
-                it.createdDate = Instant.now()
-                persist(it)
-                it
-            }
+        record.contents = contents
+                ?.filter { it.text != null }
+                ?.takeIf { it.isNotEmpty() }
+                ?.mapTo(HashSet()) { content ->
+                    RecordContent().apply {
+                        this.record = record
+                        this.fileName = content.fileName
+                        this.createdDate = Instant.now()
+                        this.contentType = content.contentType
+                        this.size = content.text!!.length.toLong()
+                        this.content = Hibernate.getLobCreator(em.get().unwrap(Session::class.java))
+                                .createBlob(content.text!!.toByteArray(Charsets.UTF_8))
+                    }
+                }
+                ?.onEach { persist(it) }
 
-            metadata = RecordMetadata().apply {
-                this.record = record
-                status = if (tmpContent == null) RecordStatus.INCOMPLETE else RecordStatus.READY
-                message = if (tmpContent == null) "No data uploaded yet" else "Data successfully uploaded, ready for processing."
-                createdDate = Instant.now()
-                modifiedDate = Instant.now()
-                revision = 1
-                persist(this)
-            }
+        record.metadata = RecordMetadata().apply {
+            this.record = record
+            status = if (metadata?.status == RecordStatus.READY && record.contents != null) RecordStatus.READY else RecordStatus.INCOMPLETE
+            message = metadata?.message ?: "Initial state"
```

Comment:
Can we make the message based on the `status` that is computed ? 
`if (status == RecordStatus.READY) "Data had been uploaded and ready for processing" else "Record has been created. Data must be uploaded"` ...
Something like this?